appeals/api/src/server/endpoints/representations/representations.service.js

```diff
@@ -96,13 +99,55 @@ export const updateRepresentationStatus = async (id, status, notes, reviewer) =>
  * @param {string} redactedRepresentation
  * @param {string} reviewer
  */
-export const redactRepresentation = async (id, redactedRepresentation, reviewer) => {
-	const rep = await representationRepository.updateRepresentationById(
-		id,
-		redactedRepresentation,
-		undefined,
-		undefined,
-		reviewer
-	);
-	return rep;
+export const redactRepresentation = (id, redactedRepresentation, reviewer) =>
+	representationRepository.updateRepresentationById(id, { redactedRepresentation, reviewer });
+
+/**
+ * @typedef {Object} CreateRepresentationInput
+ * @property {'comment' | 'statement' | 'final_comment'} representationType
+ * @property {{ firstName: string, lastName: string, email: string }} ipDetails
+ * @property {{ addressLine1: string, addressLine2?: string, town: string, county?: string, postCode: string }} ipAddress
+ * @property {string[]} attachments
+ * @property {string} redactionStatus
+ *
+ * @param {number} appealId
+ * @param {CreateRepresentationInput} input
+ * @returns {Promise<import('@pins/appeals.api').Schema.Representation>}
+ * */
+export const createRepresentation = async (appealId, input) => {
+	const { ipDetails, ipAddress } = input;
+
```

Comment:
Probably the 3 repository calls below could be wrapped in a single transaction - maybe early optimisation we don't need at this stage...

Reply:
> Probably the 3 repository calls below could be wrapped in a single transaction - maybe early optimisation we don't need at this stage...

That's a good point, I did initially consider doing it all as a single Prisma query using nested `create` fields. The problem with both approaches is it introduces references to ServiceUser and Address into `representation.repository.js`, which we might want to keep solely focused on representations.

Might be worth the trade-off though, let me know what you think.

Reply:
That's a good point, I did initially consider doing it all as a single Prisma query using nested create fields. The problem with both approaches is it introduces references to ServiceUser and Address into representation.repository.js, which we might want to keep solely focused on representations.

Might be worth the trade-off though, let me know what you think.

Reply:
Probably we can go with this and then optimise later if we need. The nested fields may work better as they would be includes to the rep repository, as the rep schema still has dependencies on those, but happy to keep it as it is for now!